src/libs/ajax/workspaces/providers/WorkspaceProvider.test.ts

```diff
@@ -0,0 +1,52 @@
+import { Ajax } from 'src/libs/ajax';
+import { asMockedFn } from 'src/testing/test-utils';
+
+import { workspaceProvider } from './WorkspaceProvider';
+
+jest.mock('src/libs/ajax');
+
+type AjaxContract = ReturnType<typeof Ajax>;
+type WorkspacesAjaxContract = Pick<AjaxContract, 'Workspaces'>['Workspaces'];
+type WorkspacesAjaxNeeds = Pick<WorkspacesAjaxContract, 'list'>;
+
+interface AjaxMockNeeds {
```

Comment:
the object/function layering makes it gnarly.... and because I want to assert on call arguments to some of those "mid-functions, this helper (fixture) pattern is working well to remove that repetition from the // Arrange sections of the tests.

Reply:
Ok. Yeah, after dealing with DataRepo's Ajax module, this makes more sense to me.